src/components/victory-gauge.js

```diff
@@ -302,75 +321,176 @@ export default class VictoryGauge extends React.Component {
       .sort(null)
       .startAngle(degreesToRadians(props.startAngle))
       .endAngle(degreesToRadians(props.endAngle))
-      .padAngle(degreesToRadians(props.padAngle))
-      .value((datum) => { return datum.y; });
+      .padAngle(degreesToRadians(props.padAngle));
+  }
+  getSegments(props) {
+    const {segments} = props;
+    if (segments && segments.length) {
+      return segments;
+    }
+    // if there are no segments provide set of data as a
+    // segment that spans entire arc. array of 1?
+    return [1];
+  }
+  getGaugeRange(props, segmentLocations) {
+    const radiansToDegrees = (r) => r * (180 / Math.PI);
+    const {domain} = props;
+    return {
+      minimum: {
+        value: domain && domain[0] || segmentLocations[0].data,
+        degrees: radiansToDegrees(segmentLocations[0].startAngle)
+      },
+      maximum: {
+        value: domain && domain[1] || segmentLocations[1].data,
+        degrees: radiansToDegrees(segmentLocations.reverse()[0].endAngle)
+      }
+    };
   }
 
   renderData(props, calculatedProps) {
-    const {style, colors, pathFunction, labelPosition, data} = calculatedProps;
+    const {
+      style, colors, pathFunction,
+      tickValues, radius,
+      segmentLocations} = calculatedProps;
+    // TODO fix data events
     const dataEvents = this.getEvents(props.events.data, "data");
+    // TODO fix label events
     const labelEvents = this.getEvents(props.events.labels, "labels");
-    const layoutFunction = this.getSliceFunction(props);
-    const slices = layoutFunction(data);
 
-    return (slices.map((slice, index) => {
-      const fill = this.getColor(style, colors, index);
-      const datum = slice.data;
-      const dataStyles = omit(slice.data, ["x", "y", "label"]);
-      const sliceStyle = defaults({}, {fill}, style.data, dataStyles);
-      const dataProps = defaults(
-        {},
-        this.getEventState(index, "data"),
-        props.dataComponent.props,
+    // figure out a way to incorperate tickcount with segments so that:
```

Comment:
You should make use of d3-scale here as in the [d3 example](http://bl.ocks.org/msqr/3202712)

like so:
```
// a linear scale that maps domain values to a percent from 0..1
scale = d3.scale.linear()
			.range([0,1])
			.domain([config.minValue, config.maxValue]);
```